### DISCUSSION TITLE
nano 5.9 missing yaml syntax highlighting

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#2333] [Everyday usage] [answered]
According to [nano](https://www.nano-editor.org/news.php) Syntax highlighting for YAML files was added in version 5.9.

Looking in `/usr/local/Cellar/nano/5.9/share/nano`

I don't see a `yaml.nanorc` file.

```
10/22/21_12:58:23 /Users/john
$ nano -V
 GNU nano, version 5.9
 (C) 1999-2011, 2013-2021 Free Software Foundation, Inc.
 (C) 2014-2021 the contributors to nano
 Compiled options: --disable-libmagic --enable-utf8

10/22/21_13:01:31 /Users/john
$ which nano
/usr/local/bin/nano

10/22/21_13:10:54 /Users/john
$ brew install nano

Warning: nano 5.9 is already installed and up-to-date.
To reinstall 5.9, run:
  brew reinstall nano

10/22/21_13:12:22 /Users/john
$ brew config
HOMEBREW_VERSION: 3.2.17-123-g96815c8
ORIGIN: https://github.com/Homebrew/brew
HEAD: 96815c8e60a1bfe398cae7c2385be5c369ff03a8
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 5448693b0d53ab053d2cfae0babd9a2e430c8ebc
Core tap last commit: 3 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--no-quarantine", "--appdir=/Applications", "--fontdir=/Library/Fonts", "--qlplugindir=/Library/QuickLook", "--internet_plugindir=/Library/Internet Plug-Ins", "--prefpanedir=/Library/PreferencePanes/", "--servicedir=/Library/Services"]
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.Ux78lq6W24/org.xquartz:0
HOMEBREW_EDITOR: nano
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 12.0.0 build 1200
Git: 2.33.1 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4
XQuartz: 2.8.1 => /opt/X11
```

**Comment by alebcay** [comment, marked as answer]
Taking a look in the `syntax` subdirectory of the 5.9 source tarball, I see these files/folders:
```
Makefile.am       email.nanorc       man.nanorc       python.nanorc
Makefile.in       extra/             markdown.nanorc  ruby.nanorc
asm.nanorc        go.nanorc          nanohelp.nanorc  rust.nanorc
autoconf.nanorc   groff.nanorc       nanorc.nanorc    sh.nanorc
awk.nanorc        guile.nanorc       nftables.nanorc  sql.nanorc
c.nanorc          html.nanorc        objc.nanorc      tcl.nanorc
changelog.nanorc  java.nanorc        ocaml.nanorc     tex.nanorc
cmake.nanorc      javascript.nanorc  patch.nanorc     texinfo.nanorc
css.nanorc        json.nanorc        perl.nanorc      xml.nanorc
default.nanorc    lua.nanorc         php.nanorc
elisp.nanorc      makefile.nanorc    po.nanorc
```

I don't see a YAML `.nanorc` file here either, so I'm not sure how it's supported in this release. In any case, Homebrew can't ship what we don't receive from upstream.

**Reply by ghost** [reply]
I get it. Just pointing it out in discussion.

**Reply by carlocab** [reply]
Yes, but what Caleb is pointing out is that this probably needs to be raised upstream for it to be fixed. Mentioning it here is unlikely to change anything.

**Reply by ghost** [reply]
I don't maintain it. Don't know a thing about it. Usually I ask in discussion before creating an issue in the hopes that someone with far more knowledge can help, which I appreciate, or if it's a verifiable issue correct it for the benefit of everyone. I don't want to clutter things up with a "non-issue" in Issues.

My response was just acknowledging his comment in the last sentence... nothing more or less. I never said I expected Homebrew to ship what it didn't receive.

**Reply by carlocab** [reply]
> I don't maintain it. Don't know a thing about it.

This much is clear. It would be very disappointing if someone for whom neither of the above was true asked this here.

> if it's a verifiable issue correct it for the benefit of everyone.

And now you know what you can do to try to help this along.

> I never said I expected Homebrew to ship what it didn't receive.

I'm not sure where I said you expected this, but ok. Thanks for sharing.

**Reply by ghost** [reply]
@carlocab Please stop hi jacking this thread to blow off your misdirected steam. 

https://github.com/Homebrew/.github/blob/365ffebb09f4b43e05d25c52973355dd0b8718dc/CODE_OF_CONDUCT.md

**Reply by carlocab** [reply]
I am merely responding to points you mentioned, so if what I said is considered highjacking this thread, perhaps that suggests something about what you said in the first place.

As for the code of conduct, I would suggest that you read it carefully for yourself and adjust your communication accordingly.

**Reply by SMillerDev** [reply]
> I don't maintain it. Don't know a thing about it. Usually I ask in discussion before creating an issue in the hopes that someone with far more knowledge can help, which I appreciate, or if it's a verifiable issue correct it for the benefit of everyone. I don't want to clutter things up with a "non-issue" in Issues.

But what exactly is the point of creating a discussion with the people who, by your own admission, can't do anything about it. Upstream doesn't provide a yaml file so they need to be told it's missing, not homebrew.

**Reply by ghost** [reply]
I was really lost over the reaction from @carlocab and @SMillerDev. Didn't understand your frustration, then I realized you were operating from a different perspective than I was.

You were probably wondering why I didn't "close" the discussion if I agreed with @alebcay? ( Which I did when I said "I get it" ). 

I didn't feel it was my place to take any action at all, especially since I've been made to feel like a fool so many times in these discussions and on other forums as well. And I guess it's true, I don't have as much knowledge. For you both this is easy and repetitive and you are very knowledgeable, don't assume everyone who pops in here has the same knowledge or wherewithal.

> But what exactly is the point of creating a discussion with the people who, by your own admission, can't do anything about it. Upstream doesn't provide a yaml file so they need to be told it's missing, not homebrew.

I am not sure where you are reading this from. I never "admitted" that people can't do anything about it ( by the way any of you could do what you **told me to do** just as easily as I could, so don't say you "can't do anything about it" ). 

I understood after it was explained to me that most likely the issue was the software from nano, and agreed with the premise of @alebcay statement, however I didn't think to close the discussion.

From my point of view, instead of being highly defensive and using your time to attack me, any one of you could have said, "It looks like it wasn't included in the software, I will notify them, or if you want to do it go ahead. Thanks!"

PS I did notify and wasn't made out to be a fool like the last time - when I was incorrect.

https://savannah.gnu.org/bugs/?61391

**Reply by carlocab** [reply]
That's how open source software works; we're all dilettantes. I'm glad you decided to get in touch with the developers. You've now helped make `nano` a little bit better for everyone who uses it.
